core/tests/dispatch_integration_tests.rs

```diff
@@ -696,45 +676,37 @@ fn remote_lost_and_dropped_connection() {
     let remote_a = system_from_network_config(NetworkConfig::default());
     let remote_port = remote_a.system_path().port();
 
-    let (ponger_named, ponf) = remote_a.create_and_register(PongerAct::new_lazy);
-    let poaf = remote_a.register_by_alias(&ponger_named, "custom_name");
-    ponf.wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    poaf.wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
+    let (ponger_named, _) = create_and_run_lazy_ponger(&remote_a);
+    remote_a
+        .register_by_alias(&ponger_named, "custom_name")
+        .wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
     let named_path = ActorPath::Named(NamedPath::with_system(
         remote_a.system_path(),
         vec!["custom_name".into()],
     ));
-    let named_path_clone = named_path.clone();
-    let named_path_clone2 = named_path.clone();
 
-    let (pinger_named, pinf) =
-        system.create_and_register(move || PingerAct::new_lazy(named_path_clone));
-    pinf.wait_expect(Duration::from_millis(1000), "Pinger failed to register!");
-
-    remote_a.start(&ponger_named);
-    system.start(&pinger_named);
-
-    // Wait for the pingpong
-    thread::sleep(Duration::from_millis(2000));
+    let (pinger_named, pinger_done_future1) = create_and_run_lazy_pinger(&system, &named_path);
+    pinger_done_future1
+        .wait_timeout(Duration::from_secs(10))
+        .expect("Pings should complete!");
 
     // Assert that things are going as we expect
-    let pongfn = remote_a.kill_notify(ponger_named);
-    pongfn
+    remote_a
+        .kill_notify(ponger_named)
         .wait_timeout(Duration::from_millis(1000))
         .expect("Ponger never died!");
+
     pinger_named.on_definition(|c| {
         assert_eq!(c.count, PING_COUNT);
     });
+
     // We now kill system2
     remote_a.kill_system().ok();
 
     // Start a new pinger on system
-    let (pinger_named2, pinf2) =
-        system.create_and_register(move || PingerAct::new_lazy(named_path));
-    pinf2.wait_expect(Duration::from_millis(1000), "Pinger failed to register!");
-    system.start(&pinger_named2);
+    let (pinger_named2, _) = create_and_run_lazy_pinger(&system, &named_path);
     // Wait for it to send its pings, system should recognize the remote address
-    thread::sleep(Duration::from_millis(1000));
+    thread::sleep(Duration::from_millis(2000));
```

Comment:
Ah yes, an arbitrary timeout increase to the rescue^^ I thought we were trying to get rid of these things?